test/cpp/jit/test_gpu.cpp

```diff
@@ -15168,6 +15190,1569 @@ TEST(NVFuserTest, FusionZeroSizeTensorNormalization_CUDA) {
+  std::vector<IValue> inputs = {t0};
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion);
+  auto outputs = fe.runFusion(inputs);
+
+  auto t1 = t0 + 1;
+  auto t2 = shift(t1, {0, 1});
+  auto t3 = shift(t1, {-1, 0});
+  auto t4 = t2 + t3;
+  auto ref = t4;
+
+  testValidate(&fusion, outputs, inputs, {ref}, __LINE__, __FILE__);
+}
+
+TEST(NVFuserTest, FusionShiftDoubleSplitMerge1_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeSymbolicTensor(2);
+  fusion.addInput(tv0);
+  auto tv1 = add(tv0, new Double(1));
+  auto tv2 = add(tv1, new Double(2));
+  auto tv3 = shift(tv2, {0, 1});
+  fusion.addOutput(tv3);
+
+  int split_factor1 = 8;
+  int split_factor2 = 4;
+
+  tv3->split(-1, split_factor1);
+
+  tv0->computeAt(tv3, -2);
+
+  tv1->split(-1, split_factor2);
+  tv1->merge(-2, -1);
+
+  // t1 and t2 allocation: (split_factor1 + 1)
+  GpuLower gpulw(&fusion);
+  for (const auto& kir_node : gpulw.kernel()->irNodes()) {
+    if (auto alloc = dynamic_cast<kir::Allocate*>(kir_node.get())) {
+      auto tensor_name = alloc->buffer()->name();
+      if (tensor_name == 1 || tensor_name == 2) {
+        TORCH_CHECK(alloc->shape().size() == 1);
+        auto def =
+            dynamic_cast<kir::BinaryOp*>(alloc->shape().at(0)->definition());
+        auto lhs = dynamic_cast<kir::Int*>(def->as<kir::BinaryOp>()->lhs());
+        TORCH_CHECK(lhs != nullptr && lhs->isConst());
+        int lhs_value = *lhs->value();
+        auto rhs = dynamic_cast<kir::Int*>(def->as<kir::BinaryOp>()->rhs());
+        TORCH_CHECK(rhs != nullptr && rhs->isConst());
+        int rhs_value = *rhs->value();
+        TORCH_CHECK(lhs_value == split_factor1 && rhs_value == 1);
+      }
+    }
+  }
+
+  FusionExecutor fe;
+  fe.compileFusion(&fusion);
+
+  int numel_x = 99;
+  int numel_y = 101;
+
+  auto options = at::TensorOptions().dtype(at::kFloat).device(at::kCUDA, 0);
+  at::Tensor t0 = at::randn({numel_x, numel_y}, options);
+  std::vector<IValue> inputs = {t0};
+  auto outputs = fe.runFusion(inputs);
+
+  auto t1 = t0 + 3;
+  auto ref = shift(t1, {0, 1});
+
+  testValidate(&fusion, outputs, inputs, {ref}, __LINE__, __FILE__);
+}
+
+TEST(NVFuserTest, FusionShiftDoubleSplitMerge2_CUDA) {
+  Fusion fusion;
+  FusionGuard fg(&fusion);
+
+  auto tv0 = makeSymbolicTensor(2);
+  fusion.addInput(tv0);
+
+  auto tv1 = add(tv0, new Double(1));
+  auto tv2 = add(tv1, new Double(2));
+  auto tv3 = shift(tv2, {1, 1});
+  fusion.addOutput(tv3);
+
+  auto out = tv3;
+
+  int split_factor1 = 32;
+  int split_factor2 = 4;
+
+  out->split(-1, split_factor1);
+  out->split(-1, split_factor2);
+  out->split(0, split_factor1);
+  out->split(1, split_factor2);
+  out->reorder({{3, 1}, {1, 2}, {4, 3}, {2, 4}});
+  out->merge(2, 3);
+  out->merge(2, 3);
+  out->merge(2, 3);
+  out->merge(0, 1);
+
+  tv0->computeAt(out, 1);
+
+  out->axis(0)->parallelize(ParallelType::BIDx);
+  out->axis(1)->parallelize(ParallelType::TIDx);
+
+  for (auto tv : {tv1, tv2}) {
```

Comment:
They work really nicely. I was able to simplify the scheduling. It may be also possible to simplify the other tests as well, which I'll do so incrementally. Thanks for the suggestion.